### DISCUSSION TITLE
[SUGGESTION] Ability to have video playing for music videos.

### DISCUSSION THREAD

**Discussion by KiimaRo** [TT-ReBORN/Georgia-ReBORN#182] [Ideas]
I'd personally love if I could have my music videos being shown with something like the youtube source component. (it works fantastic for local videos compared to the mpv component)

**Comment by TT-ReBORN** [comment]
Hi @KiimaRo,

I am not familiar with the youtube user-component ( never used it ). If it does not have an interface between the user-components controls and JavaScript, it is not possible to do so. Your only option would be to use the foo_flowin panel container.

Here is a copy and paste explanation from previous users who wanted other components
to implement directly ( without using floating popup panels ) into the theme:

> Georgia-ReBORN's layout is just one single Spider Monkey Panel where all the JavaScripts are running.
> That means when you directly insert additional panels, they will be appended to the theme and you won't be able
> to resize them individually, only when you resize the whole player.
> 
> If you want to add additional panels to the theme, you can install ttsping's foo_flowin:
> https://github.com/ttsping/foo_flowin/releases
> They are floating panels that can be resized and dragged around and you can insert your desired user-components into them.
> 
> More information can be also read here:
> https://github.com/TT-ReBORN/Georgia-ReBORN/discussions/82

**Comment by KiimaRo** [comment]
I have no idea, all I know is that I managed to get it to work on the Eole theme but I like the looks and layout a lot more on Georgia-ReBORN, but if video is not possible it's a no go for me.

For Eole I switched out the visualizer shpeck component for the Youtube source component and it worked flawlessly, was looking if I could do something similar for this but yea, I couldn't figure it out as I can't code to save my life.

**Reply by TT-ReBORN** [reply]
As mentioned in my reply, Georgia-ReBORN layout design is using only one Spider Monkey Panel for everything 
and is very different from other themes. Eole and other themes are using containers for each panel; there are pros and cons 
for both layout designs which I will not discuss any further...

You could add another Spider Monkey Panel in the Columns UI layout but it would not look good because the additional container
panel would be either on the right side or beneath Georgia-ReBORN's main panel.

**So there are only two options:** 
1. Every user can install the [foo_flowin](https://github.com/ttsping/foo_flowin/releases)  user-component, with this component you have the ability to create floating windows where you can directly add user-components, this is the fastest and easiest option.

2. If a user-component does have support ( interface ) between the component and Javascript,
    it is possible to directly implement it into Georgia-ReBORN ( main panel ), this would be the best option.
    
I hope this explains everything now.

-TT
